invenio_administration/assets/semantic-ui/js/invenio_administration/src/formik/GenerateForm.js

```diff
@@ -69,6 +70,15 @@ const mapFormFields = (obj, parentField, isCreate, formFieldsConfig, dropDumpOnl
     return null;
   }
 
+  // copy order field from config to marshmallow schema
```

Comment:
Question: Why do we need this?

Reply:
The `obj` field - is a marshmallow schema, which doesn't know anything about the order. We still have to use it in further actions, so the idea was to just copy `order` field from config there.

Reply:
Fields `order` should come from the backend (as it was before with `resourceSchema`). Conceptually, at some point the backend should deserialize `uiSchema` taking into account `ui` parameters found in `resourceSchema` (if any).

However, I do not understand what was failing here. `resourceSchema` was not passing an `ui` property anymore? That's the only reason I can see why this was failing. The sort function is conceptually the same (sorting by `>` or `-` yields the same boolean value needed by `sort`)

Reply:
Discussed IRL about this.

It seems that, [at some point](https://github.com/inveniosoftware/invenio-administration/commit/794ed8532fd02ba80126e1f479c4acfc310782c6#diff-b7a1171262f96ab548ab96789d64e3003b17f599f9a3ab8fb97a0f4781513afdL239), we separated `resourceSchema` from `uiSchema`. However, the `sort` function expects that the schema has a property `.ui` that was previously removed (see previous commit).
Moreover, the `sortFields` function reducer is wrong. It's reducing an array to an object and objects in JS are indexed by key (using its hash), therefore everything seems to be "ordered" alphabetically. Changing this means there are a couple of small changes down the  way.

Reply:
Thank you for your comments!
In the end, removed copying fields from one schema to another and changed with using necessary schema in a corresponding place: ui schema for rendering, marshmallow schema - for de/serialisation.
Thanks to @alejandromumo for support and help!